Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs

```diff
@@ -231,24 +234,47 @@ async UniTask InitializeContainersAsync(IPluginSettingsContainer settingsContain
 
             var entityParticipantTable = new EntityParticipantTable();
 
-            container.ChatMessagesBus = new DebugPanelChatMessageBus(
-                new SelfResendChatMessageBus(
-                    new MultiplayerChatMessagesBus(container.MessagePipesHub, container.ProfileRepository, new MessageDeduplication<double>()),
-                    identityCache,
-                    container.ProfileRepository
-                ),
-                debugBuilder
-            );
-
             var queuePoolFullMovementMessage = new ObjectPool<SimplePriorityQueue<NetworkMovementMessage>>(
                 () => new SimplePriorityQueue<NetworkMovementMessage>(),
                 actionOnRelease: x => x.Clear()
             );
 
+            var remoteEntities = new RemoteEntities(
+                container.RoomHub,
+                entityParticipantTable,
+                staticContainer.ComponentsContainer.ComponentPoolsRegistry,
+                queuePoolFullMovementMessage
+            );
+
+            ILoadingScreen loadingScreen = new LoadingScreen(container.MvcManager);
+
+            IRealmNavigator realmNavigator = new RealmNavigator(
+                loadingScreen,
+                mapRendererContainer.MapRenderer,
+                container.RealmController,
+                parcelServiceContainer.TeleportController,
+                container.RoomHub,
+                remoteEntities,
+                staticContainer.GlobalWorldProxy
+            );
+
+            var chatCommandsFactory = new Dictionary<Regex, Func<IChatCommand>>
+            {
+                { ChangeRealmChatCommand.REGEX, () => new ChangeRealmChatCommand(realmNavigator) },
+                { TeleportToChatCommand.REGEX, () => new TeleportToChatCommand(realmNavigator) },
+                { DebugPanelChatCommand.REGEX, () => new DebugPanelChatCommand(container.DebugContainer.Builder) },
+            };
+
+            container.ChatMessagesBus = new MultiplayerChatMessagesBus(container.MessagePipesHub, container.ProfileRepository, new MessageDeduplication<double>())
+                                       .WithSelfResend(identityCache, container.ProfileRepository)
+                                       .WithCommands(chatCommandsFactory)
+                                       .WithDebugPanel(debugBuilder);
```

Comment:
If you decided to move commands to the decorator at least get rid of `DebugPanelChatMessageBus` as we spoke about it: it does not have any behaviour that requires a decorator